internal/api/admin.go

```diff
@@ -459,7 +470,7 @@ func (a *API) adminUserCreate(w http.ResponseWriter, r *http.Request) error {
 	})
 
 	if err != nil {
-		if strings.Contains("invalid format for ban duration", err.Error()) {
+		if strings.Contains(err.Error(), "invalid format for ban duration") {
```

Comment:
found a small bug here that wasn't worth opening a separate PR to fix - `strings.Contains` takes in a string and substring and this was flipped which resulted in the error always falling through the a 500

Reply:
Can consider matching on the error code as well since it's easy. to break if the string changes. 

I think the current approach is fine though especially since it's not the focus of this PR.

Reply:
yeah that's true, i don't like the substring match either - can't rely on the error code too since we throw a generic one for validation errors

decided to move the ban duration validation out of the transaction so we don't have to handle the error matching